core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -612,16 +634,20 @@ public List<AcquiredRecords> acquire(
                     .setFirstOffset(inFlightBatch.firstOffset())
                     .setLastOffset(inFlightBatch.lastOffset())
                     .setDeliveryCount((short) inFlightBatch.batchDeliveryCount()));
+                acquiredCount += (int) (inFlightBatch.lastOffset() - inFlightBatch.firstOffset() + 1);
```

Comment:
I suppose this `acquiredCount may also be inaccurate if the topic is compacted, right?

Reply:
Yeah, but server can't determine that as we do not read the records content in server. Rather clients inform for any gaps in the record batch.